### PR TITLE
New Sync Gateway Offer

### PR DESCRIPTION
**Pull request by malscent**

### CODE REVIEW

File: gcp/couchbase-server/couchbase.py [hunk 1]
```diff
@@ -1,10 +1,28 @@
 from hashlib import sha1
 
-def get_services(version):
-    if version.startswith('7'):
-        return 'data,index,query,fts,analytics,eventing,backup'
+def get_services(context):
+    version = context.properties['serverVersion']
+    services = ''
+    if context.properties['data']:
```
Comment (tahmmee) [inline gcp/couchbase-server/couchbase.py:6]: I never understood the lack of switch in python

File: gcp/couchbase-server/couchbase.py [hunk 2]
```diff
@@ -1,10 +1,28 @@
...
+        services += 'data,'
+    if context.properties['index']:
+        services += 'index,'
+    if context.properties['query']:
+        services += 'query,'
+    if context.properties['search']:
+        services += 'fts,'
+    if context.properties['eventing']:
+        services += 'eventing,'
+    if version.startswith('7') and context.properties['backup']:
```
Comment (tahmmee) [inline gcp/couchbase-server/couchbase.py:16]: considering the future future  when 8.x is released, you may get more mileage from converting Major version to int or doing semvar compare.

**Review by tahmmee — APPROVED** [review]
you have won the marketplace

File: gcp/couchbase-server/couchbase.py (hunk 2 above)
Comment (malscent) [reply gcp/couchbase-server/couchbase.py]: Good idea.. Fixed

Comment (tahmmee) [reply gcp/couchbase-server/couchbase.py]: nice

**Review by tahmmee — APPROVED** [review]

**Pull request merged**